apicast/src/oauth/keycloak.lua

```diff
@@ -248,7 +262,8 @@ function _M:get_token(service, client)
   -- call Keycloak authorize
   local url = self.config.token_url
 
-  local res = http_client.post(url, req_body)
+  local headers = ngx.req.get_headers()
+  local res = http_client.post(url, ngx.req.get_post_args(), { headers = headers })
```

Comment:
Test backend can just expect headers that are passed. Like
```lua
test_backend.expect{ url = 'http://example.com/t', headers = { ['User-Agent'] = tostring(user_agent) } }
        .respond_with{ status = 200  }
```

IMO we should limit what is sent to Keycloak according to the spec/documentation and not whatever client sends.